Lib/test/test_compileall.py

```diff
@@ -502,19 +502,19 @@ def setUp(self):
         self.directory = tempfile.mkdtemp()
         self.source_path = os.path.join(self.directory, '_test.py')
         with open(self.source_path, 'w', encoding='utf-8') as file:
-            file.write('# -*- coding: utf-8 -*-\n')
-            file.write('print u"\u20ac"\n')
+            file.write('print(b"\u20ac")\n')  # intentional error
 
     def tearDown(self):
         shutil.rmtree(self.directory)
 
     def test_error(self):
-        try:
-            orig_stdout = sys.stdout
-            sys.stdout = io.TextIOWrapper(io.BytesIO(),encoding='ascii')
```

Comment:
The bug report was that the code failed depending on stdout encoding: https://github.com/python/cpython/issues/50965#issuecomment-1093477751

> UnicodeEncodeError occurs depending on code page.

With your change, you don't control stdout encoding anymore.